### ISSUE TITLE
eksctl encrypted AMI configuration

### ISSUE THREAD

**Issue by dayyehm** [weaveworks/eksctl#858]
Hello,
We have a scenario where a customer wants to launch node groups with encrypted root volume and therefore we attempted the following config file using eksctl : 

We attempted created an encrypted AMI from the base EKS AMI:
========================================
```YAML
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: cluster-5
  region: us-west-2

nodeGroups:
  - name: ng1-public
    instanceType: m5.xlarge
    minSize: 2
    maxSize: 8
    volumeSize: 100
    volumeType: gp2
      #Encrypted:true<<<<<<<<<<<< This option and the one below both are not recognized and they failed to work.
#Encrypted: true  <<<<<<<<<<<<<<<<<<<<<<<<<<this option caused error parsing the config file then I tried the one above it.
    ami: ami-0c7a91109d8ef0a3b  <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<< I created this encrypted AMI from the base AMI that is available in us-west-2 region
    amiFamily: AmazonLinux2
    labels:
      nodegroup-type: frontend-workloads
    iam:
      withAddonPolicies:
        autoScaler: true
        externalDNS: true
        appMesh: true
        albIngress: true
```
=======================

Upon trying that the cluster and the node group was launched but the node group Cloudformation stack was stuck creating , looking further I see that the template configuration are not correct and therefore the instances were launched and terminated ( error on launch) due to the fact that the AMI is encrypted.

Is there a way to configure the volume details via this config file to avoid modifying the Cloudformation manually after launching a non-encrypted node.

And is there a full documentation on what fields are supported/not supported via those configuration files. (like a full documentation of the API : eksctl.io/v1alpha5)

Kindly advice as soon as possible

Thank you

**Comment by errordeveloper** [comment]
Hi @dayyehm. We don't support this at the moment, so it will be a feature requests. It should relatively easy to implement.

**Comment by errordeveloper** [comment]
@adamjohnson01 do I recall correctly that you had some interest in this also?

**Comment by adamjohnson01** [comment]
@errordeveloper, not interested in this specifically but I am happy to have a go at implementing it.

**Comment by errordeveloper** [comment]
Closed via #889.
